### ISSUE TITLE
opentelemetry-api 1.42.0 pom depends on unreleased testing-internal

### ISSUE THREAD

**Issue by smola** [open-telemetry/opentelemetry-java#6693]
**Describe the bug**
The [released pom.xml](https://repo1.maven.org/maven2/io/opentelemetry/opentelemetry-api/1.42.0/opentelemetry-api-1.42.0.pom) for opentelemetry-api 1.42.0 has this new dependency:

```xml
<dependency>
  <groupId>io.opentelemetry</groupId>
  <artifactId>testing-internal</artifactId>
  <version>1.42.0</version>
  <scope>compile</scope>
  <optional>true</optional>
</dependency>
```

This breaks `mvn dependency:go-offline` of projects depending on opentelemetry-api:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.2.0:go-offline (default-cli) on project myproject: org.eclipse.aether.resolution.DependencyResolutionException: The following artifacts could not be resolved: io.opentelemetry:testing-internal:jar:1.42.0 (absent): Could not find artifact io.opentelemetry:testing-internal:jar:1.42.0 in central (https://repo.maven.apache.org/maven2) -> [Help 1]
```

**Steps to reproduce**
1. Create project depending on opentelemetry-api 1.42.0.
2. Run  `mvn dependency:go-offline`

**What did you expect to see?**
No error on `mvn dependency:go-offline`.

**What did you see instead?**
The error described above.

**What version and what artifacts are you using?**
Artifacts: opentelemetry-api
Version: 1.42.0
How did you reference these artifacts? (excerpt from your `build.gradle`, `pom.xml`, etc):

```
        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-api</artifactId>
            <version>LATEST</version>
        </dependency>
```

**Environment**
Compiler: Primarily Temurin, various versions. Example Docker image: `maven:3.9-eclipse-temurin-11`.
OS: Ubuntu-based docker images.
Runtime (if different from JDK above): See above.
OS (if different from OS compiled on): See above.

**Additional context**

**Comment by BartoszBrodzinski-TomTom** [comment]
I also have this problem in my repositories

**Comment by jack-berg** [comment]
<img width="856" alt="Screenshot 2024-09-09 at 9 36 06 AM" src="https://github.com/user-attachments/assets/ee84f8d1-0a31-4459-93d0-a00a86f566a2">

And also `juni:junit`, `org.assertj:assertj-core`, and `org.mockito:mockito-core`. 😞 

Seems to be the result of these new `textFixtures` dependencies: https://github.com/open-telemetry/opentelemetry-java/blob/b0643fef389d26702c10b67fd04ffc2fc5aedaad/api/all/build.gradle.kts#L21-L24

Will look into it and almost certainly publish a patch release.

**Comment by jack-berg** [comment]
Released a patch for this in [1.42.1](https://github.com/open-telemetry/opentelemetry-java/releases/tag/v1.42.1).

Thanks for reporting and sorry about that.
